sdk/nodejs/runtime/resource.ts

```diff
@@ -154,6 +154,50 @@ export function readResource(res: Resource, t: string, name: string, props: Inpu
     }), label);
 }
 
+// The shape of data returned by the built-in invoke to `pulumi:pulumi:readStackResource`.
+interface StackResourceResult {
+    urn: string;
+    // The output properties of the resource represented by `urn` as registered in the Pulumi
+    // engine.
+    outputs: Record<string, any>;
+}
+
+/**
+ * Gets an existing custom resource's state from the engine.  Assumes that the resource has already
+ * been registered by some other code, and looks it up by URN.
+ */
+export function getResource(res: Resource, t: string, name: string, custom: boolean, props: Inputs, opts: ResourceOptions): void {
+    const urn: Input<URN> | undefined = opts.urn;
+    if (!urn) {
+        throw new Error("Cannot get resource whose options are lacking a URN value");
+    }
```

Comment:
this is a bug in our code right?

Reply:
Mostly - yes - though users can technically call `getResource` directly if they really want.  This should be unreachable when `getResource` is called through the `Resource` constructor.